modules/dicom_archive/php/dicomarchiveanonymizer.class.inc

```diff
@@ -53,13 +53,22 @@ class DICOMArchiveAnonymizer implements Mapper
         $newrow = json_decode($resource->toJSON(), true);
         $cid    = $resource->getCenterID();
 
-        if (!preg_match($config['patientNameRegex'], $newrow['Patient Name'])
-            && !preg_match($config['LegoPhantomRegex'], $newrow['Patient Name'])
-            && !preg_match($config['LivingPhantomRegex'], $newrow['Patient Name'])
+        if (!preg_match(
+            "/{$config['patientNameRegex']}/",
```

Comment:
Bad things can happen here (and other places) if the config value has a backslash. It needs to be escaped.

Reply:
sure, but would you mind elaborating? if you escape backslashes then this would disallow for any reserved regex tokens to be used wouldn't it?

Reply:
If someone enters a backslash anywhere in the config value then this will get directly interpolated based on the user data. ie if I enter the value `/data/` the value passed to preg_match will be `//data//' The matching part of the regex is then "//", with "data//" as the modifier, which wasn't the intended behaviour. I don't think a malicious user can do anything worse with this than they previously could (and since it comes from an admin in the config module is probably relatively safe), but this could generate a lot of false positive matches and/or PHP warnings.